share/functions/fish_fossil_prompt.fish

```diff
@@ -4,48 +4,88 @@ function fish_fossil_prompt --description 'Write out the fossil prompt'
         return 1
     end
 
-    # Bail if not a fossil checkout
-    if not fossil ls &> /dev/null
-	return 127
-    end
+    # Read branch and bookmark (bail if not checkout)
+    set -l branch (fossil branch current 2>/dev/null)
+    or return 127
+
+    set -g fish_color_fossil_clean green
+    set -g fish_color_fossil_modified yellow
+    set -g fish_color_fossil_dirty red
+
+    set -g fish_color_fossil_added green
+    set -g fish_color_fossil_renamed magenta
+    set -g fish_color_fossil_missing red
+    set -g fish_color_fossil_deleted red
+    set -g fish_color_fossil_untracked yellow
+    set -g fish_color_fossil_conflict red
+
+    set -g fish_prompt_fossil_status_added '✚'
+    set -g fish_prompt_fossil_status_modified '*'
+    set -g fish_prompt_fossil_status_renamed '⇒'
+    set -g fish_prompt_fossil_status_deleted '-'
+    set -g fish_prompt_fossil_status_missing '✖'
+    set -g fish_prompt_fossil_status_untracked '?'
+    set -g fish_prompt_fossil_status_conflict '×'
+
+    set -g fish_prompt_fossil_status_order added modified renamed deleted missing untracked conflict
```

Comment:
Setting these variables this way means there is absolutely no way they can be overridden, which kinda defeats the point of variables.

You probably want

```fish
set -q fish_prompt_fossil_status_order
or set -g fish_prompt_fossil_status_order added modified renamed deleted missing untracked conflict
```